src/VisualStudio/Core/SolutionExplorerShim/AnalyzersCommandHandler.cs

```diff
@@ -159,7 +154,7 @@ private void UpdateMenuItemVisibility()
                                                 Path.GetExtension(itemName).Equals(".ruleset", StringComparison.OrdinalIgnoreCase);
         }
 
-        private void UpdateMenuItemsChecked()
+        private void UpdateSeverityMenuItemsChecked()
         {
             _setSeverityErrorMenuItem.Checked = AnyDiagnosticsWithSeverity(ReportDiagnostic.Error);
             _setSeverityWarningMenuItem.Checked = AnyDiagnosticsWithSeverity(ReportDiagnostic.Warn);
```

Comment:
Looks like we will show the check if any diagnostic has matching severity. Shouldn't it be the opposite (i.e. show check only if all selected diagnostics have that severity?)